src/main/java/net/imglib2/algorithm/gradient/PartialDerivative.java

```diff
@@ -234,4 +234,142 @@
 			}
 		}
 	}
+	
+	/**
+	 * Compute the backward difference of source in a particular dimension.
+	 * 
+	 * @param source source image, has to provide valid data in the interval of
+	 *            the gradient image plus a one pixel border in dimension.
+	 * @param gradient output image
+	 * @param dim along which dimension the partial derivatives are computed
+	 */
+	public static <T extends NumericType<T>> void gradientBackwardDifference(final RandomAccessible<T> source,
+		final RandomAccessibleInterval<T> gradient, final int dim)
+	{
+		final int n = gradient.numDimensions();
+
+		final long[] min = new long[n];
+		gradient.min(min);
+		final long[] max = new long[n];
+		gradient.max(max);
+		final long[] shiftback = new long[n];
+		for (int d = 0; d < n; ++d)
+			shiftback[d] = min[d] - max[d];
+
+		final RandomAccess<T> result = gradient.randomAccess();
+		final RandomAccess<T> back = source.randomAccess(Intervals.translate(
+			gradient, 1, dim));
+		final RandomAccess<T> current = source.randomAccess(gradient);
+
+		result.setPosition(min);
+		back.setPosition(min);
+		back.bck(dim);
+		current.setPosition(min);
+
+		final long max0 = max[0];
+		while (true) {
+			// process pixel
+			final T t = result.get();
+			t.set(current.get());
+			t.sub(back.get());
+
+			// move to next pixel
+			// check dimension 0 separately to avoid the loop over d in most
+			// iterations
+			if (result.getLongPosition(0) == max0) {
+				if (n == 1) return;
+				result.move(shiftback[0], 0);
+				back.move(shiftback[0], 0);
+				current.move(shiftback[0], 0);
+				// now check the remaining dimensions
+				for (int d = 1; d < n; ++d)
+					if (result.getLongPosition(d) == max[d]) {
+						result.move(shiftback[d], d);
+						back.move(shiftback[d], d);
+						current.move(shiftback[d], d);
+						if (d == n - 1) return;
+					}
+					else {
+						result.fwd(d);
+						back.fwd(d);
+						current.fwd(d);
+						break;
+					}
+			}
+			else {
+				result.fwd(0);
+				back.fwd(0);
+				current.fwd(0);
+			}
+		}
+	}
+	
+	/**
+	 * Compute the forward difference of source in a particular dimension.
```

Comment:
Please at some JavaDoc math that specifies exactly what is being computed, like
https://en.wikipedia.org/wiki/Finite_difference#Forward,_backward,_and_central_differences

Reply:
Having the math in the doc would be very helpful, either in code
```
d_f( x ) = f( x + h ) - f( x )
```
or tex
[![](https://wikimedia.org/api/rest_v1/media/math/render/svg/9b4b1b463ac6a532d0089dad4ef7f7ec1a91acb9)](https://en.wikipedia.org/wiki/Finite_difference#Forward,_backward,_and_central_differences)
(examples for forward difference).